api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java

```diff
@@ -121,6 +128,293 @@ private CohortDefinition getTransferredInPatients() {
+   *             consultation between Diagnosis Date and Diagnosis Date+7days as following:ve who
+   *             have the first consultation registered in “Ficha Clinica” (encounter type 6) after
+   *             “Data Diagnostico” with the following conditions:
+   *             <ul>
+   *               <li>[encounter type 6] “Data da consulta” (encounter datetime) >= [encounter type
+   *                   53] “Data Diagnóstico” (concept_id 22772 obs datetime) and <= [encounter type
+   *                   53] “Data Diagnóstico” (concept_id 22772 obs datetime) + 7 days
+   *             </ul>
+   *       </ul>
+   *   <li>D: Filter all child patients who have the first clinical consultation between Diagnosis
+   *       Date and Diagnosis Date+7days as following:
+   *       <ul>
+   *         <li>all patients who have the first consultation registered in “Ficha Clinica”
+   *             (encounter type 6) after “Data Diagnostico” as following:
+   *             <ul>
+   *               <li>
+   *                   <p>Select the oldest date between “Data Diagnóstico” (concept_id 23807 obs
+   *                   datetime, encounter type 53) and “Data Diagnóstico Presuntivo” (concept_id
+   *                   22772 obs datetime, encounter type 53)
+   *                   <p>(Note: if “Data Diagnóstico” is empty then consider “Data Diagnóstico
+   *                   Presuntivo” if it exists. If “Data Diagnostico” is empty then consider “Data
+   *                   Diagnostico Presuntivo” if it exists).
+   *             </ul>
+   *         <li>And the first consultation [encounter type 6] “Data da consulta” (encounter
+   *             datetime) >= [encounter type 53] the oldest “Data Diagnóstico” minus the oldest
+   *             “Data Diagnóstico” is >=0 and <=7 days
+   *       </ul>
+   * </ul>
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getMQC3N1() {
+    CompositionCohortDefinition compositionCohortDefinition = new CompositionCohortDefinition();
+
+    compositionCohortDefinition.setName("Melhoria de Qualidade Category 3 Numerator");
+
+    compositionCohortDefinition.addParameter(new Parameter("startDate", "startDate", Date.class));
+    compositionCohortDefinition.addParameter(new Parameter("endDate", "endDate", Date.class));
+    compositionCohortDefinition.addParameter(new Parameter("location", "location", Date.class));
+
+    compositionCohortDefinition.addSearch(
+        "CAT3DEN", EptsReportUtils.map(this.getMQC3D1(), MAPPING));
+
+    compositionCohortDefinition.addSearch("C", EptsReportUtils.map(this.getCFromMQC3N1(), MAPPING));
+    compositionCohortDefinition.addSearch("D", EptsReportUtils.map(this.getDFromMQC3N1(), MAPPING));
+
+    compositionCohortDefinition.setCompositionString("CAT3DEN AND (C OR D)");
+
+    return compositionCohortDefinition;
+  }
+
+  private CohortDefinition getCFromMQC3N1() {
```

Comment:
Might be a cleaner alternative to remove this method since the dataset already applies the dis aggregation between children and adults, no?

Reply:
yes @amosh101  if you read the spec will find that  C and D  are applied to filter adults and child respectively.